src/sentry/relay/config/metric_extraction.py

```diff
@@ -247,20 +224,28 @@ def _trim_if_above_limit(
     project: Project,
     widget_type: str,
 ) -> list[HashedMetricSpec]:
-    """If we have more specs than the max limit we should trim it."""
-    return_specs = list(specs)
-
-    if len(specs) > max_specs:
-        # Do not log for Sentry
-        if project.organization.id != 1:
-            logger.error(
-                "Too many (%s) on demand metric %s for project %s",
-                len(specs),
-                widget_type,
-                project.slug,
-            )
+    """Trim specs per version if above max limit"""
+    return_specs = []
+    specs_per_version: dict[int, list[HashedMetricSpec]] = {}
+    for hash, spec, spec_version in specs:
+        specs_per_version.setdefault(spec_version.version, [])
+        specs_per_version[spec_version.version].append((hash, spec, spec_version))
+
+    for version, specs_for_version in specs_per_version.items():
+        if len(specs_for_version) > max_specs:
+            # Do not log for Sentry
+            if project.organization.id != 1:
```

Comment:
I need to add an option to control this (self-hosted folks would hit this) or also check for the host or check for the slug.